csi/controller.go

```diff
@@ -1115,6 +1115,36 @@ func (s *OsdCsiServer) ListSnapshots(
 	ctx context.Context,
 	req *csi.ListSnapshotsRequest,
 ) (*csi.ListSnapshotsResponse, error) {
+	// Get any labels passed in by the CO
+	_, locator, _, err := s.specHandler.SpecFromOpts(req.GetSecrets())
```

Comment:
Why do we need to call `specHandler.SpecFromOpts` here? Why not just look at the map returned by `req.GetSecrets()` and check if it contains `osdSnapshotCredentialIDKey` ?